app/src/main/java/org/mozilla/fenix/StrictModeManager.kt

```diff
@@ -70,19 +103,30 @@ class StrictModeManager(config: Config) {
     /**
      * Runs the given [functionBlock] and sets the given [StrictMode.ThreadPolicy] after its
      * completion when in a build configuration that has StrictMode enabled. If StrictMode is
-     * not enabled, simply runs the [functionBlock].
-     *
-     * This function is written in the style of [AutoCloseable.use].
+     * not enabled, simply runs the [functionBlock]. This function is written in the style of
+     * [AutoCloseable.use].
      *
-     * This is significantly less convenient to run than when it was written as an extension function
-     * on [StrictMode.ThreadPolicy] but I think this is okay: it shouldn't be easy to ignore StrictMode.
+     * This function contains perf improvements so it should be
+     * called instead of [mozilla.components.support.ktx.android.os.resetAfter] (using the wrong
+     * method should be prevented by a lint check). This is significantly less convenient to run than
+     * when it was written as an extension function on [StrictMode.ThreadPolicy] but I think this is
+     * okay: it shouldn't be easy to ignore StrictMode.
      *
      * @return the value returned by [functionBlock].
      */
     inline fun <R> resetAfter(policy: StrictMode.ThreadPolicy, functionBlock: () -> R): R {
         // Calling resetAfter takes 1-2ms (unknown device) so we only execute it if StrictMode can
         // actually be enabled. https://github.com/mozilla-mobile/fenix/issues/11617
         return if (isEnabledByBuildConfig) {
+            // This can overflow and crash. However, it's unlikely we'll suppress StrictMode 9
+            // quintillion times in a build config where StrictMode is enabled so we don't handle it
+            // because it'd increase complexity.
+            suppressionCount += 1
+            logger.warn("StrictMode violation suppressed: #$suppressionCount")
+            if (Thread.currentThread() == mainLooper.thread) { // markers only supported on main thread.
+                components.core.engine.profiler?.addMarker("StrictMode.suppression", "Count: $suppressionCount")
```

Comment:
Just to make sure, are we adding a marker to be able to see the count # on the profiler? 

If so, I think we should document that somewhere to highlight Logcat vs marker logging. I don't think adding a comment here would be worthwhile since it isn't the goal of the method. However, I could see a someone coming across this code and wondering why we're using `profiler?.addMarker` when we use `logger.warn` right before. 

I'm not sure if my thought made sense, I can try to re-explain it if it didn't!

Reply:
> Just to make sure, are we adding a marker to be able to see the count # on the profiler?

I thought it would be useful, when we're analyzing start up performance, to be able to see all of the places that we access the disk or the network on the main thread.

> I'm not sure if my thought made sense, I can try to re-explain it if it didn't!

@MarcLeclair could you elaborate further? Maybe it'd be good to talk about in stand up. I honestly don't know how I feel about both adding a marker and logging yet. The decision to do both was confusing for me.

Reply:
@mcomella  I think that having both statement ```logger.warn``` and ```profiler?.addMarker``` can be confusing for someone looking at how we did things. I was wondering if we should document somewhere why we used the marker API here along with the Logger.

Reply:
Sure, I'll add a comment.